lib/ohloh_scm/system.rb

```diff
@@ -32,6 +32,12 @@ def temp_folder
       ENV['OHLOH_SCM_TEMP_FOLDER_PATH'] || Dir.tmpdir
     end
 
+    def repository_temp_folder
+      temp_path = (ENV['OHLOH_SCM_TRASH_PATH'] || Dir.tmpdir) + "/#{Time.now.utc.strftime('%d%b%y')}"
+      FileUtils.mkdir_p(temp_path)
+      FileUtils.mkdir_p(temp_path + "/#{DateTime.now.strftime('%Q')}").first
```

Comment:
Shouldn't mkdir_p here take care of a scenario where **temp_path** is missing? So the line above this is not needed.